qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +99,343 @@ class Command(BaseModel):
+        """
+        Return a list of the qubits for which the given operation is defined. Single qubit
+        operations return a flat list, and multiqubit operations return a list of ordered tuples.
+
+        Args:
+            operation: The name of the circuit operation.
+
+        Returns:
+            Qubit indices which have the given operation defined. This is a list of tuples if the
+            operation has an arity greater than 1, or a flat list of ints otherwise.
+        """
+        return [qubits[0] if len(qubits) == 1 else qubits
+                for qubits in sorted(self._ops_def[operation].keys())]
+
+    def qubit_ops(self, qubits: Union[int, Iterable[int]]) -> List[str]:
+        """
+        Return a list of the operation names that are defined by the backend for the given qubit
+        or qubits.
+
+        Args:
+            qubits: A qubit index, or a list or tuple of indices.
+
+        Returns:
+            All the operations which are defined on the qubits. For 1 qubit, all the 1Q
+            operations defined. For multiple qubits, all the operations which apply to that
+            whole set of qubits (e.g. qubits=[0, 1] may return ['cx']).
+        """
+        return self._qubit_ops[_to_tuple(qubits)]
+
+    def has(self, operation: str, qubits: Union[int, Iterable[int]]) -> bool:
+        """
+        Is the operation defined for the given qubits?
+
+        Args:
+            operation: The operation for which to look.
+            qubits: The specific qubits for the operation.
+
+        Returns:
+            True iff the operation is defined.
+        """
+        return operation in self._ops_def and \
+            _to_tuple(qubits) in self._ops_def[operation]
+
+    def assert_has(self, operation: str, qubits: Union[int, Iterable[int]]) -> None:
+        """
+        Convenience method to check that the given operation is defined, and error if it is not.
+
+        Args:
+            operation: The operation for which to look.
+            qubits: The specific qubits for the operation.
+
+        Returns:
+            None
+
+        Raises:
+            PulseError: If the operation is not defined on the qubits.
+        """
+        if not self.has(operation, _to_tuple(qubits)):
+            if operation in self._ops_def:
+                raise PulseError("Operation '{op}' exists, but is only defined for qubits "
+                                 "{qubits}.".format(op=operation,
+                                                    qubits=self.qubits_with_op(operation)))
+            raise PulseError("Operation '{op}' is not defined for this "
+                             "system.".format(op=operation))
+
+    def get(self,
+            operation: str,
+            qubits: Union[int, Iterable[int]],
+            *params: List[Union[int, float, complex]],
+            **kwparams: Dict[str, Union[int, float, complex]]) -> Schedule:
```

Comment:
adding spurious parameters does not raise an error here, but it probably should
```defaults.get('u3', [0], 0.1, 0.2, 0.3, 0.4, 0.5)```

Reply:
On that note, why are the parameters not just a list here?
```defaults.get('u3', [0], [0.1, 0.2, 0.3])```

this is how the circuit instructions are created for example. 
```Instruction(name='myinst', num_qubits=1, num_clbits=0, params=[0.1,0.2,0.3])```

When using it I have found it a bit surprising that the qubits have to be list (e.g. in the case of 'cx') but not the params.

Reply:
because parameters are expected to be kwargs, such as `P0`, `P1`, `P2` ..., to be conformant to pulse Qobj. when they are given by args, `.get` method implicitly links the args to command's `._params` (List[str]) to make a dictionary of parameters based on the order of each item. if we can assume a parameter naming/ordering rule, `params` can be an array.